### ISSUE TITLE
执行 sh scripts/car/vob/eval_with_trained_model.sh 卡住

### ISSUE THREAD

**Issue by Jianqiang-Mei** [zju3dv/disprcnn#37]
OFFLINE_2D_PREDICTIONS: models/kitti/car/vob/mask/inference/kitti_%s_vob_car/predictions
  OPTIMIZER: SGD
  PRINT_INTERVAL: 1
  PRINT_ITERATION: False
  REMOVE_EMPTY_ANNOTATIONS_TRAIN: False
  SCHEDULER: OneCycleScheduler
  STEPS: (30000,)
  TRAIN_2D: True
  TRAIN_PC: False
  TRAIN_PSM: True
  UNCERT_LOSS_WEIGHT: 0
  WARMUP_FACTOR: 0.3333333333333333
  WARMUP_ITERS: 500
  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0001
  WEIGHT_DECAY_BIAS: 0
TEST:
  DETECTIONS_PER_IMG: 100
  EXPECTED_RESULTS: []
  EXPECTED_RESULTS_SIGMA_TOL: 4
  FORCE_RECOMPUTE: True
  IMS_PER_BATCH: 1
Loading PSMNet from models/kitti/car/vob/idispnet/idispnet.pth
Loading PSMNet from models/kitti/car/vob/idispnet/idispnet.pth
2022-01-13 14:44:23,459 disprcnn.utils.checkpoint INFO: Loading checkpoint from models/kitti/car/vob/idispnet/idispnet.pth
Loading PSMNet from models/kitti/car/vob/idispnet/idispnet.pth
Loading PSMNet from models/kitti/car/vob/idispnet/idispnet.pth
  0%|                                                                                                              | 0/1885 [00:00<?, ?it/s]2022-01-13 14:44:27,048 disprcnn.inference INFO: Start evaluation on kitti_val_vob_car dataset(3769 images).
100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1885/1885 [03:26<00:00,  9.11it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1885/1885 [03:36<00:00,  8.70it/s]
2022-01-13 14:48:03,729 disprcnn.inference INFO: Total run time: 0:03:36.680315 (0.11498026823801398 s / img per device, on 2 devices)
2022-01-13 14:48:03,729 disprcnn.inference INFO: Model inference time: 0:02:56.793129 (0.09381434290565335 s / img per device, on 2 devices)

**Comment by f-sky** [comment]
卡在哪里？接下来本应该是什么？

**Comment by 13998697003mjq** [comment]
> 卡在哪里？接下来本应该是什么？

这是eval_with_trained_model.sh里的内容：
python -m torch.distributed.launch --nproc_per_node $NGPUS tools/test_net.py \
--config-file configs/kitti/car/vob/mask.yaml --ckpt models/kitti/car/vob/mask/smrcnn.pth \
DATASETS.TEST "('kitti_val_vob_car',)"

python -m torch.distributed.launch --nproc_per_node $NGPUS tools/test_net.py \
--config-file configs/kitti/car/vob/idispnet.yaml --ckpt models/kitti/car/vob/idispnet/idispnet.pth \
MODEL.DISPNET.TRAINED_MODEL "models/kitti/car/vob/idispnet/idispnet.pth" \
DATASETS.TEST "('kitti_val_vob_car',)"

python -m torch.distributed.launch --nproc_per_node $NGPUS tools/test_net.py \
--config-file configs/kitti/car/vob/rcnn.yaml \
--ckpt models/kitti/car/vob/rcnn/pointrcnn.pth \
DATASETS.TEST "('kitti_val_vob_car',)"

我觉得应该是卡在了第二条指令，就是evaluate idispnet的时候，
如果执行顺利，应该像第一条指令一样，执行完之后给出evaluate的结果，像这样：

  WARMUP_METHOD: linear
  WEIGHT_DECAY: 0.0005
  WEIGHT_DECAY_BIAS: 0
TEST:
  DETECTIONS_PER_IMG: 100
  EXPECTED_RESULTS: []
  EXPECTED_RESULTS_SIGMA_TOL: 4
  FORCE_RECOMPUTE: True
  IMS_PER_BATCH: 8
2022-01-14 16:21:39,213 disprcnn.utils.checkpoint INFO: Loading checkpoint from models/kitti/car/vob/mask/smrcnn.pth
2022-01-14 16:21:45,317 disprcnn.data.build WARNING: When using more than one image per GPU you may encounter an out-of-memory (OOM) error if your GPU does not have sufficient memory. If this happens, you can reduce SOLVER.IMS_PER_BATCH (for training) or TEST.IMS_PER_BATCH (for inference). For training, you must also adjust the learning rate and schedule length according to the linear scaling rule. See for example: https://github.com/facebookresearch/Detectron/blob/master/configs/getting_started/tutorial_1gpu_e2e_faster_rcnn_R-50-FPN.yaml#L14
When using more than one image per GPU you may encounter an out-of-memory (OOM) error if your GPU does not have sufficient memory. If this happens, you can reduce SOLVER.IMS_PER_BATCH (for training) or TEST.IMS_PER_BATCH (for inference). For training, you must also adjust the learning rate and schedule length according to the linear scaling rule. See for example: https://github.com/facebookresearch/Detectron/blob/master/configs/getting_started/tutorial_1gpu_e2e_faster_rcnn_R-50-FPN.yaml#L14
2022-01-14 16:21:49,195 disprcnn.inference INFO: Start evaluation on kitti_val_vob_car dataset(3769 images).
  0%|                                                                                                               | 0/236 [00:00<?, ?it/s]/home/mjq/myproject/disprcnn/disprcnn/structures/boxlist_ops.py:94: UserWarning: This overload of nonzero is deprecated:
        nonzero()
Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  /pytorch/torch/csrc/utils/python_arg_parser.cpp:882.)
  (ws >= min_size) & (hs >= min_size)
/home/mjq/myproject/disprcnn/disprcnn/structures/boxlist_ops.py:94: UserWarning: This overload of nonzero is deprecated:
        nonzero()
Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  /pytorch/torch/csrc/utils/python_arg_parser.cpp:882.)
  (ws >= min_size) & (hs >= min_size)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 236/236 [06:39<00:00,  1.69s/it]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 236/236 [06:40<00:00,  1.70s/it]
2022-01-14 16:28:30,219 disprcnn.inference INFO: Total run time: 0:06:41.024051 (0.212801300584242 s / img per device, on 2 devices)
2022-01-14 16:28:30,219 disprcnn.inference INFO: Model inference time: 0:03:52.498657 (0.12337418758891311 s / img per device, on 2 devices)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 3769/3769 [00:00<00:00, 4616.24it/s]
-----using iou thresh0.7------
mkdir: 无法创建目录"models/kitti/car/vob/mask/inference/kitti_val_vob_car/txt/plot": 文件已存在
Loading detections...
  done.
Starting 2D evaluation (car) ...
save models/kitti/car/vob/mask/inference/kitti_val_vob_car/txt/plot/car_detection.txt
0.984012 0.894281 0.802180
sh: 1: gnuplot: not found
sh: 1: gnuplot: not found
Error: /undefinedfilename in (car_detection.eps)
Operand stack:

Execution stack:
   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push
Dictionary stack:
   --dict:960/1684(ro)(G)--   --dict:0/20(G)--   --dict:77/200(L)--
Current allocation mode is local
Last OS error: No such file or directory
GPL Ghostscript 9.26: Unrecoverable error, exit code 1
sh: 1: pdfcrop: not found
  done.
-----using iou thresh0.5------
mkdir: 无法创建目录"models/kitti/car/vob/mask/inference/kitti_val_vob_car/txt/plot": 文件已存在
Loading detections...
  done.
Starting 2D evaluation (car) ...
save models/kitti/car/vob/mask/inference/kitti_val_vob_car/txt/plot/car_detection.txt
0.992258 0.905176 0.814129
sh: 1: gnuplot: not found
sh: 1: gnuplot: not found
Error: /undefinedfilename in (car_detection.eps)
Operand stack:

Execution stack:
   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push
Dictionary stack:
   --dict:960/1684(ro)(G)--   --dict:0/20(G)--   --dict:77/200(L)--
Current allocation mode is local
Last OS error: No such file or directory
GPL Ghostscript 9.26: Unrecoverable error, exit code 1
sh: 1: pdfcrop: not found
  done.
0.7
AP 2d 98.40 89.43 80.22
0.5
AP 2d 99.23 90.52 81.41

**Comment by f-sky** [comment]
你可以一条条的运行bash里的命令。遇到卡住的情况可以ctrl+c中断掉。有可能已经跑完了但没print出来，参考[这里](https://github.com/zju3dv/disprcnn/blob/master/docs/TRAIN_VAL.md#notes).
